ADRs/0011-Python-Path-Resolution.md

```diff
@@ -0,0 +1,322 @@
+
+## Proposal
+The base of the proposal is to be able to assign identification tokens to different python install locations. Instead of knowing and specifying absolute python path locations, the user can just specify the identification token of system found python installs. There is still going to be options available for specifying a custom python install that are not present in system environment **PATH** variable. 
+
+### The `pythonpaths` Command
+An example of such a workflow is as follows: 
+
+```shell script
+konduit pythonpaths
+```  
+
+which will output the python install locations as follows:
+
+```text
+-------------------------------------------PYTHON INSTALLS-------------------------------------------
+-   id: 1
+    location: C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\python.exe
+    version: 3.7.8
+
+-   id: 2
+    location: C:\Users\shams\miniconda3\python.exe
+    version: 3.7.7
+
+-   id: 3   
+    location: C:\Users\shams\AppData\Local\Microsoft\WindowsApps\python.exe                        
+    version: 3.6.1
+-----------------------------------------------------------------------------------------------------
+
+--------------------------------------------CONDA INSTALLS-------------------------------------------
+-   id: 1
+    location: C:\Users\shams\miniconda3\Scripts\conda.exe
+    version: 4.8.4
+    ----------------------------------------ENVIRONMENTS---------------------------------------------
+    -   name: base
+        location: C:\Users\shams\miniconda3
+        version: 3.6.1
+    
+    -   name: py37
+        location: C:\Users\shams\miniconda3\envs\py37
+        version: 3.7.7
+    
+    -   name: py38
+        location: C:\Users\shams\miniconda3\envs\py38
+        version: 3.8.1
+    -------------------------------------------------------------------------------------------------
+
+-   id: 2
+    location: C:\Users\shams\miniconda2\Scripts\conda.exe
+    version: 4.7.1
+    ----------------------------------------ENVIRONMENTS---------------------------------------------
+    -   name: base
+        location: C:\Users\shams\miniconda2
+        version: 2.6.1
+    
+    -   name: py27
+        location: C:\Users\shams\miniconda2\envs\py27
+        version: 2.7.7
+    
+    -   name: py25
+        location: C:\Users\shams\miniconda2\envs\py25
+        version: 2.5.1
+    -------------------------------------------------------------------------------------------------
+-----------------------------------------------------------------------------------------------------
+```
+
+Based on the above info (gained from the command: `konduit pythonpaths`), it's much easier to specify python paths in configuration files (using `type`, `location`, `environment` properties):
+
+```json
+{
+  "@type" : "PYTHON",
+  "pythonConfig" : {
```

Comment:
Could you show a code example as well? It'd be good to see something like pythonPathProfileId(..) as well.

Specify basic validation as well like: "couldn't find path.." etc

Reply:
Sure. I'm calling it location for now for the custom python type where `location=E:\python37\python.exe` could make more sense

Reply:
I'll add an example end-to-end workflow

Reply:
I've changed `location` to just `path` in the examples. They're not exactly end-to-end but they give you the whole workflow idea pretty well.